src/data-gateway/search-data-gateway.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  EVENTS_FILE_ENV,
+  NEWS_FILE_ENV,
+  PROGRAM_FILE_ENV,
+  PROJECT_FILE_ENV,
+} from 'config';
+import { type SearchDetailInput } from 'entities';
+import { type IFileService } from 'services/file-services/flat-file-types';
+import { type ISearchDataGateway } from 'usecases';
+
+const fileName = [
+  NEWS_FILE_ENV,
+  EVENTS_FILE_ENV,
+  PROGRAM_FILE_ENV,
+  PROJECT_FILE_ENV,
+];
+
+export class SearchDataGateway implements ISearchDataGateway {
+  constructor(private readonly fileService: IFileService) {}
+
+  async fetch(data: SearchDetailInput): Promise<any[]> {
+    const fileNameCategory = this.getFileNameForCategory(data.category);
+
+    try {
+      var results: any[] = [];
+      for (const fileName of fileNameCategory) {
+        const currentContent = await this.fileService.read(fileName);
+
+        if (!currentContent) {
+          continue;
+        }
+
+        const parsedData = JSON.parse(currentContent);
+
+        parsedData.filter((item) => {
+          for (const key in item) {
+            if (
+              typeof item[key] === 'string' &&
+              item[key].toLowerCase().includes(data.searchTerm)
+            ) {
+              item.category = fileName.split('.')[0].toUpperCase();
+              results.push(item);
+              return true;
+            }
+          }
+          return false;
+        });
+      }
+      return results;
+    } catch (error) {
+      return [];
+    }
+  }
+
+  private getFileNameForCategory(category: string): string {
+    const categoryToMap = {
+      news: [NEWS_FILE_ENV],
+      events: [EVENTS_FILE_ENV],
+      projects: [PROJECT_FILE_ENV],
+      programs: [PROGRAM_FILE_ENV],
+    };
+
+    return categoryToMap[category] || fileName;
```

Comment:
The logic of this search needs to not know where the data is being stored. 

In other words, we should depend on the data gateways to get these values